### PR TITLE
New config!

### PR DESCRIPTION
**Pull request by mitchellhamilton**
~~Still need to write tests for this.~~

### CODE REVIEW

File: packages/config/src/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+let unsafeParse = parse as any;
+
+describe("parser errors", () => {
+  test("changelog invalid value", () => {
+    expect(() => {
+      unsafeParse({ changelog: {} });
+    }).toThrowErrorMatchingInlineSnapshot(`
+"Some errors occurred when validating the changesets config:
+The \`changelog\` option is set as {} when the only valid values are undefined, a module path or a tuple with a module path and config for the changelog generator"
```
Comment (Noviny) [inline packages/config/src/index.test.ts]: Lovely seeing just all the error messages. Now I understand why you want an error message format guide.

This is the only one I'm not sure on. Both tuple and module path might confuse newer programmers, but also, that's just what they are? Could we include examples of each to make this cognitively easier to parse?

File: packages/config/src/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    linked: [],
+    changelog: false,
+    commit: true,
+    access: "private"
+  });
+});
+
+let defaults = {
+  linked: [],
+  changelog: ["@changesets/cli/changelog", null],
```
[the remark this answers is not in the document]
Comment (mitchellhamilton) [reply packages/config/src/index.test.ts]: I suppose we could check if the repo is a github or bitbucket repo and ask the user what they want and then install the appropriate package when changeset init is run.

File: packages/config/schema.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      "type": "boolean",
+      "description": "Determines whether changesets should commit the results of the add and bump command"
+    },
+    "access": {
+      "enum": ["private", "public"],
+      "type": "string",
+      "description": "Determines whether changesets should publish packages to the registry publicly or privately"
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (mitchellhamilton) [reply packages/config/schema.json]: It looks like default values are in the JSON schema spec, will add them https://json-schema.org/understanding-json-schema/reference/generic.html

[the remark this answers is not in the document]
Comment (mitchellhamilton) [reply packages/config/schema.json]: Also, it seems like we can provide example which would probably be good for the changelog option but let's add those when we have other changelog generator packages.

[the remark this answers is not in the document]
Comment (mitchellhamilton) [reply packages/config/schema.json]: Also, just tested the things in VS Code and the default/examples are shown in autocomplete

**Comment by mitchellhamilton** [comment]
@Noviny updated All The Things

**Pull request merged**